test/integration/logs/benchmark/common_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package benchmark
+
+import (
+	"flag"
+	"io"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+
+	logsjson "k8s.io/component-base/logs/json"
+	"k8s.io/klog/v2"
+)
+
+func init() {
+	// Cause all klog output to be discarded with minimal overhead.
+	// We don't include time stamps and caller information.
+	klog.InitFlags(nil)
+	flag.Set("alsologtostderr", "false")
+	flag.Set("logtostderr", "false")
+	flag.Set("skip_headers", "true")
+	flag.Set("one_output", "true")
+	flag.Set("stderrthreshold", "FATAL")
+	klog.SetOutput(&output)
+}
+
+type bytesWritten int64
+
+func (b *bytesWritten) Write(data []byte) (int, error) {
+	l := len(data)
+	*b += bytesWritten(l)
+	return l, nil
+}
+
+func (b *bytesWritten) Sync() error {
+	return nil
+}
+
+var output bytesWritten
+var jsonLogger = newJSONLogger(&output)
+
+func newJSONLogger(out io.Writer) logr.Logger {
+	encoderConfig := &zapcore.EncoderConfig{
```

Comment:
Any reason this config doesn't resemble more the real configuration?

Looks like this skips caller info and uses different encoders than normal JSON formatter. Is this intentional?

Reply:
Oh, I didn't review commit by commit so I didn't read whole commit messages.

Reply:
To elaborate further, stack unwinding and timestamping adds noise that I didn't want when comparing the encoding aspect. It also makes it hard to verify output.

If we were to enable it, the performance numbers would be questionable because the call site is always the same. I don't know how much that affects performance, but it didn't seem useful.